.github/workflows/ci.yml

```diff
@@ -0,0 +1,372 @@
+            -D selinux=false
+
+        # Disable all options to compile-test their fallbacks. Run the test
+        # suite to run basic fallback verification.
+        - id: "fallback"
+          name: "Ubuntu-x86_64-clang-release"
+
+          cc: "clang"
+          cc_ld: "lld"
+          image: "ghcr.io/readaheadeu/rae-ci-ubuntu:latest"
+          setupargs: >-
+            -D apparmor=false
+            -D audit=false
+            -D docs=false
+            -D launcher=false
+            -D selinux=false
+
+        # A reduced build with `-O0` to verify we do not rely on dead-code
+        # elimination.
+        - id: "O0-PLAIN"
+          name: "Ubuntu-x86_64-gcc-plain"
+
+          buildtype: "plain"
+          cc: "gcc"
+          cc_ld: "bfd"
+          image: "ghcr.io/readaheadeu/rae-ci-ubuntu:latest"
+          setupargs: >-
+            --buildtype plain
+            -D c_args=-O0
+
+        # An aggressive -O3 -DNDEBUG build that verfies that we properly
+        # follow strict aliasing rules and do not rely on debug builds.
+        - id: "O3-NDEBUG"
+          name: "Ubuntu-x86_64-gcc-ndebug"
+
+          cc: "gcc"
+          cc_ld: "bfd"
+          image: "ghcr.io/readaheadeu/rae-ci-ubuntu:latest"
+          setupargs: >-
+            --buildtype release
+            -D b_ndebug=true
+            -D optimization=3
+            -D audit=true
+            -D launcher=true
+
+        # Run with LTO and all possible optimizations.
+        - id: "lto"
+          name: "Ubuntu-x86_64-clang-release"
+
+          cc: "clang"
+          cc_ld: "lld"
+          image: "ghcr.io/readaheadeu/rae-ci-ubuntu:latest"
+          setupargs: >-
+            --buildtype release
+            -D b_lto=true
+            -D b_lto_mode=default
+            -D b_ndebug=true
+            -D optimization=3
+
+        # Run with thinLTO and all possible optimizations.
+        - id: "thinlto"
+          name: "Ubuntu-x86_64-clang-release"
+
+          cc: "clang"
+          cc_ld: "lld"
+          image: "ghcr.io/readaheadeu/rae-ci-ubuntu:latest"
+          setupargs: >-
+            --buildtype release
+            -D b_lto=true
+            -D b_lto_mode=thin
+            -D b_ndebug=true
+            -D optimization=3
+
+        # A release build with `--cross-file lib32` to test on 32-bit.
+        - id: "32bit"
+          name: "Arch-i686-clang-release"
+
+          cc: "clang"
+          cc_ld: "lld"
+          image: "ghcr.io/readaheadeu/rae-ci-archlinux:latest"
+          rustflags: "--target i686-unknown-linux-gnu"
+          setupargs: >-
+            --cross-file lib32
+            -D apparmor=false
+            -D audit=false
+            -D selinux=false
+
+        # Run with address-sanitizers enabled.
+        - id: "asan"
+          name: "Ubuntu-x86_64-clang-release"
+
+          cc: "clang"
+          cc_ld: "lld"
+          image: "ghcr.io/readaheadeu/rae-ci-ubuntu:latest"
+          setupargs: >-
+            -D b_sanitize=address
```

Comment:
`-D b_sanitize=address` affects the C part only. https://doc.rust-lang.org/beta/unstable-book/compiler-flags/sanitizer.html#addresssanitizer would have to be passed via `RUSTFLAGS`. It's in nightly only though.